frontend/pkg/ocm/ocm.go

```diff
@@ -0,0 +1,148 @@
+package ocm
+
+import (
+	cmv1 "github.com/openshift-online/ocm-sdk-go/clustersmgmt/v1"
+	v1 "github.com/openshift/api/config/v1"
+
+	"github.com/Azure/ARO-HCP/internal/api"
+	"github.com/Azure/ARO-HCP/internal/api/arm"
+)
+
+const (
+	csCloudProvider    string = "azure"
+	csProductId        string = "aro"
+	csHypershifEnabled bool   = true
+	csMultiAzEnabled   bool   = true
+	csCCSEnabled       bool   = true
+)
+
+// ConvertCStoFrontend converts a CS Cluster object into HCPOpenShiftCluster object
+func ConvertCStoFrontend(cluster *cmv1.Cluster) (*api.HCPOpenShiftCluster, error) {
+	hcpcluster := &api.HCPOpenShiftCluster{
+		TrackedResource: arm.TrackedResource{
+			Location: "",
+			Tags:     nil,
+			Resource: arm.Resource{
+				ID:   cluster.ID(),
+				Name: cluster.Name(),
+				Type: cluster.Flavour().ID(),
+				SystemData: &arm.SystemData{
+					CreatedBy:          "",
+					CreatedByType:      "",
+					CreatedAt:          nil,
+					LastModifiedBy:     "",
+					LastModifiedByType: "",
+					LastModifiedAt:     nil,
+				},
+			},
+		},
+		Properties: api.HCPOpenShiftClusterProperties{
+			Spec: api.ClusterSpec{
+				Version: api.VersionProfile{
+					ID:                cluster.Version().ID(),
+					ChannelGroup:      cluster.Version().ChannelGroup(),
+					AvailableUpgrades: cluster.Version().AvailableUpgrades(),
+				},
+				DNS: api.DNSProfile{
+					BaseDomain:       cluster.DNS().BaseDomain(),
+					BaseDomainPrefix: cluster.DomainPrefix(),
+				},
+				Network: api.NetworkProfile{
+					NetworkType: api.NetworkType(cluster.Network().Type()),
+					PodCIDR:     cluster.Network().PodCIDR(),
+					ServiceCIDR: cluster.Network().ServiceCIDR(),
+					MachineCIDR: cluster.Network().MachineCIDR(),
+					HostPrefix:  int32(cluster.Network().HostPrefix()),
+				},
+				Console: api.ConsoleProfile{
+					URL: cluster.Console().URL(),
+				},
+				API: api.APIProfile{
+					URL:        cluster.API().URL(),
+					IP:         "",
+					Visibility: api.Visibility(cluster.API().Listening()),
+				},
+				FIPS:                          cluster.FIPS(),
+				EtcdEncryption:                cluster.EtcdEncryption(),
+				DisableUserWorkloadMonitoring: cluster.DisableUserWorkloadMonitoring(),
+				Proxy: api.ProxyProfile{
+					HTTPProxy:  cluster.Proxy().HTTPProxy(),
+					HTTPSProxy: cluster.Proxy().HTTPSProxy(),
+					NoProxy:    cluster.Proxy().NoProxy(),
+					TrustedCA:  cluster.AdditionalTrustBundle(),
+				},
+				Platform: api.PlatformProfile{
+					ManagedResourceGroup:   cluster.Azure().ManagedResourceGroupName(),
+					SubnetID:               cluster.Azure().SubnetResourceID(),
+					OutboundType:           "",
+					NetworkSecurityGroupID: cluster.Azure().NetworkSecurityGroupResourceID(),
+					EtcdEncryptionSetID:    "",
+				},
+				IssuerURL: "",
+				ExternalAuth: api.ExternalAuthConfigProfile{
+					Enabled:       false,
+					ExternalAuths: []*v1.OIDCProvider{},
+				},
+				Ingress: []*api.IngressProfile{},
```

Comment:
Similar question: was `Ingress` intentionally skipped?
Ingress profiles are simple enough that we should just do it now.

Reply:
I can get this added

Reply:
Ive updated this to empty values right now as it may not be as one-to-one as i hoped. in CS its a list of Ingresses that we'd have to loop through, but URL and IP dont appear to be available values, only Visibility (in the form of ListeningMethod) so this might need to be revisted. Unless i totally missed something!